src/devices/lacrosse_ws7000.c

```diff
@@ -0,0 +1,282 @@
+/** @file
+    LaCrosse WS7000/WS2500 weather sensors.
+
+    Copyright (C) 2019 ReMiOS and Christian W. Zuckschwerdt <zany@triq.net>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+/** @fn int lacrosse_ws7000_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+LaCrosse WS7000/WS2500 weather sensors.
+
+- WS2500-19 brightness sensor
+- WS7000-20 meteo sensor (temperature/humidity/pressure)
+- WS7000-16 Rain Sensor
+- WS7000-15 wind sensor
+
+PWM 400 us / 800 us with fixed bit width of 1200 us.
+Messages are sent as nibbles (4 bits) with LSB sent first.
+A frame is composed of a preamble followed by nibbles (4 bits) separated by a 1-bit.
+
+Message Layout:
+
+    P P S A D..D X C
+
+- Preamble: 10x bit "0", bit "1"
+- Sensor Type:  Value 0..9 determing the sensor type
+  - 0 = WS7000-27/28 Thermo sensor ( interval 177s - Adr * 0.5s )
+  - 1 = WS7000-22/25 Thermo/Humidity sensor ( interval 177s - Adr * 0.5s )
+  - 2 = WS7000-16 Rain sensor ( interval 173s - Adr * 0.5s )
+  - 3 = WS7000-15 Wind sensor ( interval 169s - Adr * 0.5s )
+  - 4 = WS7000-20 Thermo/Humidity/Barometer sensor ( interval 165s - Adr * 0.5s )
+  - 5 = WS2500-19 Brightness sensor ( interval 161s - Adr * 0.5s )
+- Address:  Value 0..7 for the sensor address
+    In case of a negative temperature the MSB of the Address becomes "1"
+- Data:     Nibbles with BCD encoded sensor data values.
+- XOR:      Nibble holding XOR of the S ^ A ^ Data-nibbles
+- Checksum: Sum of all nibbles + 5 ( S + A + nibble(0) + .. + nibble(n) + XOR + 5 ) & 0xF
+
+*/
```

Comment:
Maybe if we could add a link to the documentation here? Or maybe just add that to the tests repo text description.